CardParts/src/Resources/AssetManager.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  AssetManager.swift
+//  CardParts
+//
+//  Created by Eude K Lesperance on 10/18/19.
+//
+
+import UIKit
+
+class AssetManager {
+    static let shared = AssetManager()
+    
+    private let cache: NSCache = NSCache<NSString, UIImage>()
+    
+    init() {
+        // render and cache the default size images
+        Icon.allCases.forEach { icon in _ = image(for: icon)}
+    }
+    
+    func image(for icon: Icon, in rect: CGRect = CGRect(x: 0, y: 0, width: 24.0, height: 24.0)) -> UIImage {
+        let key = "\(icon)_\(rect.width)x\(rect.height)" as NSString
+        if let cachedImage = cache.object(forKey: key) {
+            return cachedImage
+        }
+        
+        let image = render(icon, in: rect)
+        cache.setObject(image, forKey: key)
+        return image
+    }
+    
+    private func render(_ icon: Icon, in rect: CGRect) -> UIImage {
+        let format = UIGraphicsImageRendererFormat.default()
+        let renderer = UIGraphicsImageRenderer(size: rect.size, format: format)
+        
+        let image = renderer.image { ctx in
+            ctx.cgContext.scaleBy(x: rect.width / 24.0, y: rect.height / 24.0)
+            let path = icon.bezierPath()
+            if icon == .confetti {
+                UIColor.white.setStroke()
+                path.stroke()
+            } else {
+                UIColor.white.setFill()
+                path.fill()
+            }
+        }
+        
+        return image.withRenderingMode(.alwaysTemplate)
+    }
+}
+
+extension AssetManager {
+    enum Icon: CaseIterable {
+        case arrowDown, star, diamond, triangle, pencil, confetti
+        
+        func bezierPath() -> UIBezierPath {
+            switch self {
+            case .arrowDown: return arrowDown()
+            case .star: return star()
+            case .diamond: return diamond()
+            case .triangle: return triangle()
+            case .pencil: return pencil()
+            case .confetti: return confetti()
+            }
+        }
+        
+        private func arrowDown() -> UIBezierPath {
+            let path = UIBezierPath()
+            path.move(to: CGPoint(x: 7.41, y: 8.59))
+            path.addLine(to: CGPoint(x: 12, y: 13.17))
+            path.addLine(to: CGPoint(x: 16.59, y: 8.59))
+            path.addLine(to: CGPoint(x: 18, y: 10))
+            path.addLine(to: CGPoint(x: 12, y: 16))
+            path.addLine(to: CGPoint(x: 6, y: 10))
+            path.addLine(to: CGPoint(x: 7.41, y: 8.59))
+            path.close()
+            return path
+        }
+        
+        private func star() -> UIBezierPath {
+            let size = CGSize(width: 24.0, height: 24.0)
+            let sides = 5
+            let radius: CGFloat = min(size.width, size.height) / 2.0
```

Comment:
Potentially might want to refactor this since similar vars are being declared in `confetti()` and `triangle()` 🤔